spacy/ml/models/entity_linker.py

```diff
@@ -99,13 +99,44 @@ def empty_kb_factory(vocab: Vocab):
     return empty_kb_factory
 
 
+@registry.misc("spacy.EmptyKB.v2")
+def empty_kb_for_config() -> Callable[[Vocab, int], KnowledgeBase]:
+    def empty_kb_factory(vocab: Vocab, entity_vector_length: int):
+        return InMemoryLookupKB(vocab=vocab, entity_vector_length=entity_vector_length)
+
+    return empty_kb_factory
+
+
+def get_candidates(kb: KnowledgeBase, mention: Span) -> Iterable[Candidate]:
```

Comment:
Moved this and `get_candidates_all()` here to avoid circular dependency (`candidate.py` <-> `kb.py`) - I don't think this matters?